### PR TITLE
PI-22232  Configure BasicAuth Filter supplier in SDK and update BasicAuthSecurityConfiguration

### PR DESCRIPTION
**Pull request by nsadaphal**
#### Description
1. Using OAuth2 API's for Basic Auth
  Event API's
  Domain Association API's
  Migration API's
  Restriction API's
2. Updating BasicAuthSecurityConfiguration to correct use of user details for authorization.
3. Integrating  BasicAuthSecurityConfiguration to ConnectorSdkConfiguration.
4. Adding authentication provider in BasicAuthSecurityConfiguration

#### Manual merge checklist:
- [ ] Code Review completed
- [ ] Code coverage
- [ ] QA Validation completed
  - Testrail TestCase/Plan link:
- [ ] Approved by a PI tech lead
- [ ] Github checks all green

#### Dev Testing
**Basic Auth for Event API:-**

**1. With Valid Credential:-**
Screenshot 2021-03-10 at 6 48 26 PM

Screenshot 2021-03-10 at 6 48 39 PM

**2. With InValid Credential:-**

Screenshot 2021-03-10 at 7 24 02 PM

Screenshot 2021-03-10 at 7 23 53 PM

**OAuth2 for Event API:-**

**1. With Valid Credential:-**

Screenshot 2021-03-10 at 6 52 09 PM

Screenshot 2021-03-10 at 6 52 24 PM

**2. With InValid Credential:-**

Screenshot 2021-03-10 at 7 28 06 PM

**Tested for Internal API**
**1. With OAuth2:**

Screenshot 2021-03-10 at 6 50 45 PM

Screenshot 2021-03-10 at 6 50 56 PM

**2. With Basic Auth:**
Screenshot 2021-03-10 at 7 29 33 PM

Screenshot 2021-03-10 at 7 29 47 PM

### CODE REVIEW

**Review by ketanmulay-appdirect — DISMISSED** [review]
LGTM

File: src/main/java/com/appdirect/sdk/web/oauth/BasicAuthSecurityConfiguration.java [hunk 1]
```diff
@@ -13,128 +13,86 @@
...
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .antMatchers("/unsecured/**")
+                .permitAll()
+                .and()
+                .requestMatchers()
+                .antMatchers("/api/v2/basic/integration/**", "/api/v2/basic/domainassociation/**", "/api/v2/basic/migration/**", "/api/v2/basic/restrictions/**")
```
Comment (raushan-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/BasicAuthSecurityConfiguration.java:89]: We don't need separate endpoints and configuration to support basic auth, we can just add a basic auth filter to the main configuration and existing APIs will support basic as well with OAuth1 and OAuth2

Let's have a discussion offline

Comment (nsadaphal) [reply src/main/java/com/appdirect/sdk/web/oauth/BasicAuthSecurityConfiguration.java:89]: Sure, Just adding a basic auth filter in the main configuration will not make it work, We require other beans and configurations for basic auth to work.
Also by keeping configure separate we can have clean and readable code.

Comment (nsadaphal) [reply src/main/java/com/appdirect/sdk/web/oauth/BasicAuthSecurityConfiguration.java]: @raushan-appdirect As discussed updated the implementation for basic auth to use the same Oauth2 endpoints.
Thanks

File: src/main/java/com/appdirect/sdk/web/oauth/model/SessionRequestMatcher.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+public class SessionRequestMatcher implements RequestMatcher {
+
+    private static final String AUTHORIZATION = "Authorization";
+    private static final String AD_AUTHORIZATION = "AD-Authorization";
+    private static final String BASIC = "Basic ";
+    private static final String INTEGRATION_PATH = "/api/v2/integration/**";
+    private static final String DOMAIN_ASSOCIATION_PATH = "/api/v1/domainassociation/**";
+    private static final String MIGRATION_PATH = "/api/v1/migration/**";
+    private static final String RESTRICTIONS_PATH = "/api/v1/restrictions/**";
```
Comment (raushan-appdirect) [inline src/main/java/com/appdirect/sdk/web/oauth/model/SessionRequestMatcher.java:20]: I guess here the three APIs should be v2

**Review by raushan-appdirect — CHANGES_REQUESTED** [review]
Could you please add some UTs

File: src/main/java/com/appdirect/sdk/web/oauth/model/SessionRequestMatcher.java (hunk 2 above)
Comment (nsadaphal) [reply src/main/java/com/appdirect/sdk/web/oauth/model/SessionRequestMatcher.java]: updated

**Review by raushan-appdirect — APPROVED** [review]
💯

**Review by ketanmulay-appdirect — APPROVED** [review]
LGTM

**Pull request merged**